ApiFirst/ApiFirst/Controllers/CostumersController.cs

```diff
@@ -0,0 +1,89 @@
+﻿using DomainModels.Models;
+using DomainServices.Interfaces;
+using Microsoft.AspNetCore.Mvc;
+
+namespace WebApi.Controllers
+{
+    [Route("api/[controller]")]
+    [ApiController]
+    public class CustomersController : ControllerBase
+    {
+        private readonly ICustomerRepository _service;
+
+        public CustomersController(ICustomerRepository repo)
+        {
+            _service = repo ?? throw new ArgumentNullException(nameof(repo));
+        }
+
+        [HttpGet("get-by-cpf")]
+        public Customer? GetByCpf([FromQuery] string cpf)
```

Comment:
1. É redundante ter get na rota porque o endpoint já é get
2. É um parâmetro obrigatório então ser enviado via rota e não query

```suggestion
        [HttpGet("cpf/{cpf}")]
        public Customer? GetByCpf(string cpf)
```